src/sentry/silo/patches/silo_aware_transaction_patch.py

```diff
@@ -1,29 +1,45 @@
-from typing import Optional
+from typing import Any, Callable, Optional
 
 from django import get_version
 from django.db import router, transaction
+from django.db.backends.base.base import BaseDatabaseWrapper
 from django.db.transaction import Atomic
 
 _default_atomic_impl = transaction.atomic
+_default_on_commit = transaction.on_commit
+_default_get_connection = transaction.get_connection
 
 
 class MismatchedSiloTransactionError(Exception):
     pass
 
 
 def siloed_atomic(using: Optional[str] = None, savepoint: bool = True) -> Atomic:
+    using = determine_using_by_silo_mode(using)
+    return _default_atomic_impl(using=using, savepoint=savepoint)
+
+
+def siloed_get_connection(using: Optional[str] = None) -> BaseDatabaseWrapper:
+    using = determine_using_by_silo_mode(using)
+    return _default_get_connection(using=using)
+
+
+def siloed_on_commit(cb: Callable[..., Any], using: Optional[str] = None) -> None:
+    using = determine_using_by_silo_mode(using)
+    return _default_on_commit(cb, using)
+
```

Comment:
Oh nice! I didn't realize we needed to patch these as well, but it makes sense. Are there any other transaction methods we may need to worry about other devs using?

Reply:
I *believe* no, because all the other transaction utilities receive their connection via `get_connection` anyways.